C/CFITSIO/build_tarballs.jl

```diff
@@ -3,42 +3,51 @@
 using BinaryBuilder
 
 name = "CFITSIO"
-version = v"4.4.0"
+version = v"4.6.2"
 
 # Collection of sources required to build CFITSIO
 sources = [
     ArchiveSource("http://heasarc.gsfc.nasa.gov/FTP/software/fitsio/c/cfitsio-$(version).tar.gz",
-                  "95900cf95ae760839e7cb9678a7b2fad0858d6ac12234f934bd1cb6bfc246ba9"),
-    DirectorySource("./bundled"),
+                  "66fd078cc0bea896b0d44b120d46d6805421a5361d3a5ad84d9f397b1b5de2cb"),
+    # DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/cfitsio*
-atomic_patch -p1 ../patches/configure_in.patch
-atomic_patch -p1 ../patches/Makefile_in.patch
-if [[ $target == *-freebsd* ]]; then
-   # `gethostbyname` is considered outdated and not available any more; declare it manually
-   atomic_patch -p1 ../patches/gethostbyname.patch
-fi
-autoreconf
+
 if [[ "${target}" == *-mingw* ]]; then
     # This is ridiculous: when CURL is enabled, CFITSIO defines a macro,
     # `TBYTE`, that has the same name as a mingw macro.  Let's rename all
     # `TBYTE` to `_TBYTE`.
     sed -i 's/\<TBYTE\>/_TBYTE/g' $(grep -lr '\<TBYTE\>')
 fi
 
-./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --enable-reentrant
-make -j${nproc} shared
-make install
-# Delete the static library
-rm ${prefix}/lib/libcfitsio.a
-# On Windows platforms, we need to move our .dll files to bin
-if [[ "${target}" == *-mingw* ]]; then
-    mkdir -p ${libdir}
-    mv ${prefix}/lib/*.dll ${libdir}/.
+options=(
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_FIND_ROOT_PATH=${prefix}
+    -DCMAKE_SYSROOT=/opt/${target}/${target}/sys-root
+    -DCMAKE_SYSTEM_LIBRARY_PATH=/opt/${target}/${target}/sys-root/usr/lib64/lp64d
```

Comment:
These should be handled already in the toolchain file that is passed below. Did you see an error that needed these specifically?

Reply:
Is there something we can do in BinaryBuilderBase to make the job easier?

Reply:
For example in https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/be6a14c6dd412549bae717e0f8d038720e88ea9a/src/BuildToolchains.jl#L47-L115

Reply:
Yes, I think that would make sense. Can we set `CMAKE_SYSTEM_LIBRARY_PATH` there?

Reply:
That's what a toolchain file is for (as suggested by Ian above): not to have to specify system stuff in every build 🙂

Reply:
Yes, we should be able to. It just will need another branch for riscv specifically.